extensions/funqy/funqy-knative-events/runtime/src/main/java/io/quarkus/funqy/knative/events/CloudEvent.java

```diff
@@ -1,19 +1,30 @@
 package io.quarkus.funqy.knative.events;
 
 import java.time.OffsetDateTime;
+import java.util.Map;
 
 /**
- * Cloud event. Represents only the headers. No data.
+ * CloudEvent.
  *
  */
-public interface CloudEvent {
+public interface CloudEvent<T> {
+
     String id();
 
     String specVersion();
 
     String source();
 
+    String type();
+
     String subject();
```

Comment:
I thinking of making those fields (which are optional in spec) `java.util.Optional<>` wdyt @patriot1burke @mswiderski ?

Reply:
It will cause allocation of yet another object, but it would make optionality of those fields more clear.

Reply:
Personally never really liked Optional.  Too overkill for me.  Us old guys have been doing if (foo == null) for years.

Reply:
especially fro this case where people will most likely just fill the important to them attributes and rely on defaults for other so having optional is not needed